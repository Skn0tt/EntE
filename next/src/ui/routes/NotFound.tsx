/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import { Grid, Typography } from "@material-ui/core";
import withErrorBoundary from "../hocs/withErrorBoundary";
import { Center } from "../components/Center";

const Diggie = require("../assets/Diggie.png");

export const NotFound: React.SFC<{}> = () => (
  <Center>
    <Grid item>
      <img src={Diggie} height={400} />
    </Grid>
    <Grid item>
      <Typography variant="h6">Die Seite wurde nicht gefunden.</Typography>
    </Grid>
  </Center>
);

export default withErrorBoundary()(NotFound);
