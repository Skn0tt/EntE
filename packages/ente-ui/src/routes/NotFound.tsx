/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import { Grid, Typography } from "@material-ui/core";
import withErrorBoundary from "../hocs/withErrorBoundary";

const Diggie = require("../assets/Diggie.png");

export const NotFound: React.SFC<{}> = () => (
  <Grid container alignItems="center" justify="center">
    <Grid item>
      <img src={Diggie} height={400} />
    </Grid>
    <Grid item>
      <Typography variant="h6">Die Seite wurde nicht gefunden.</Typography>
    </Grid>
  </Grid>
);

export default withErrorBoundary()(NotFound);
