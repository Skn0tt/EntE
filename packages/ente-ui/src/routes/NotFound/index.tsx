/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import { Grid, Typography, WithStyles, withStyles } from "@material-ui/core";
import styles from "./styles";
import withErrorBoundary from "../../components/withErrorBoundary";

/**
 * # Assets
 */
const Diggie = require("../../res/img/Diggie.png");

/**
 * # Component Types
 */
type Props = WithStyles;

/**
 * # Component
 */
export const NotFound: React.SFC<Props> = props => (
  <Grid container alignItems="center" justify="center">
    <Grid item>
      <img src={Diggie} height={400} />
    </Grid>
    <Grid item>
      <Typography variant="h6">Die Seite wurde nicht gefunden.</Typography>
    </Grid>
  </Grid>
);

export default withStyles(styles)(withErrorBoundary()(NotFound));
