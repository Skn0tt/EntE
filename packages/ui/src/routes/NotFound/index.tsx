/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import { Grid, Typography, WithStyles, withStyles } from "material-ui";
import styles from "./styles";

/**
 * # Assets
 */
const Diggie = require("../../res/img/Diggie.png");

/**
 * # Component Types
 */
type Props = WithStyles;

/**
 * # Component
 */
export const NotFound: React.SFC<Props> = props => (
  <Grid container alignItems="center" justify="center">
    <Grid item>
      <img src={Diggie} height={400} />
    </Grid>
    <Grid item>
      <Typography variant="title">Die Seite wurde nicht gefunden.</Typography>
    </Grid>
  </Grid>
);

export default withStyles(styles)(NotFound);
