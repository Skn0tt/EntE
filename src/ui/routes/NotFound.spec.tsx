/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import { NotFound } from "./NotFound";
import { shallow } from "enzyme";

describe("NotFound", () => {
  const comp = shallow(<NotFound />);

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });
});
