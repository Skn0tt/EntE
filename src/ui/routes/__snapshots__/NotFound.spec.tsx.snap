// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly 1`] = `
<Center>
  <WithStyles(Grid)
    item={true}
  >
    <img
      height={400}
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Typography)
      variant="h6"
    >
      Die Seite wurde nicht gefunden.
    </WithStyles(Typography)>
  </WithStyles(Grid)>
</Center>
`;
